pyuvdata/cst_beam.py

```diff
@@ -55,21 +55,24 @@ def read_cst_beam(self, filename, beam_type='power', feed_pol='x',
             self.history += self.pyuvdata_version_str
 
         if beam_type is 'power':
-            self.set_power()
             self.Naxes_vec = 1
+
+            if feed_pol is 'x':
+                feed_pol = 'xx'
+            elif feed_pol is 'y':
+                feed_pol = 'yy'
+
             if rotate_pol:
-                if feed_pol is 'x':
-                    self.polarization_array = np.array([-5, -6])
-                else:
-                    self.polarization_array = np.array([-6, -5])
+                rot_pol_dict = {'xx': 'yy', 'yy': 'xx', 'xy': 'yx', 'yx': 'xy'}
+                pol2 = rot_pol_dict[feed_pol]
+                self.polarization_array = np.array([uvutils.polstr2num(feed_pol),
+                                                    uvutils.polstr2num(pol2)])
```

Comment:
So the read function in cst_beam.py can only take a single value in feed_pol, not a list. You can pass a list to the function in uvbeam.py if you have several files, but then it internally loops over the call to cst_beam.py and only passes one value (and there's checking that that value isn't a list, tuple or array). Also, the way it's used as a key to a dict I think it would fail with a list.

Reply:
Oh, and I should have said that if you pass a list of feed_pols and files to the uvbeam.py method, it sets the rotate keyword to False by default. You could override that, but it would error when it tried to combine the files because they'd have overlapping data.